visualstimulation/utils.py

```diff
@@ -1,12 +1,67 @@
-import os
 import quantities as pq
-import exdir
 import numpy as np
 import neo
+import os
+
+
+def generate_gradiently_weighed_data(data, weight_start=1, weight_end=0.5):
+    """
+    Creates weighed data using gradients from weight_start to weight_end.
+
+    Example
+    -------
+    >>> A = np.array([2, 5, 7, 4, 8, 10, 3, 7, 3, 5])
+    >>> generate_gradiently_weighed_data(data=A)
+    array([2.        , 4.72222222, 6.22222222, 3.33333333, 6.22222222,
+          7.22222222, 2.        , 4.27777778, 1.66666667, 2.5       ])
+
+    Note
+    -------
+    In the example above the following weights are generated:
+    array([1.        , 0.94444444, 0.88888889, 0.83333333, 0.77777778,
+           0.72222222, 0.66666667, 0.61111111, 0.55555556, 0.5       ])
+
+    Parameters
+    ----------
+    data : numpy.array
+        0D numpy.array with data to be weighed
+    weight_start : int, float
+        Initial weight
+    weight_end : int, float
+        Last weight
+    Returns
+    ------
+    out : numpy.array(Weighed data); data * weight
+    """
+    if not isinstance(data, type(np.empty(0))):
+        msg = "data has to be numpy.array, and not {}".format(type(data))
+        raise TypeError(msg)
+
+    weights = np.linspace(weight_start, weight_end, len(data))
+    weighed_data = data * weights
+    return weighed_data
+
+
+def minmax_scale(data):
+    """
+    Transforms features by scaling each feature to [0, 1]
+
+    Parameters
+    ----------
+    data : numpy.array
+        0D numpy.array with data to be weighed
+    Returns
+    ------
+    out : minmax scaled ([0, 1]) numpy.array
+    """
+    if not isinstance(data, type(np.empty(0))):
+        msg = "data has to be numpy.array, and not {}".format(type(data))
+        raise TypeError(msg)
+    return (data - data.min()) / (data.max() - data.min())
```

Comment:
Note that, if `data` is a quantity array then this operation will remove the unit. Consider adding the following:

```
if isinstance(data, pq.Quantity):
      return (data - data.min()) / (data.max() - data.min()) * data.units
```

Reply:
Oh, I didn't know that. Shame that it doesn't include units for algebreic operations.

Note that the normalised data is a ratio, and all ratios are dimensionless. The user of the function should note the units in the plot in the xlable/ylable.

I agree that we should include the suggested function in case the normalised data is used for further computation.

Reply:
Well, it does not remove the unit (I was imprecise), but makes it dimensionless which is technically correct so the array will still be a quantity array.